src/box/xrow_update_field.c

```diff
@@ -693,9 +728,13 @@ xrow_update_string_store_leaf(const char *data, size_t size, void *cb_arg)
 }
 
 static char *
-xrow_update_store_string(const struct xrow_update_string *str, char *out)
+xrow_update_store_strbin(const struct xrow_update_string *str, char *out,
+			 bool is_string)
```

Comment:
So you propose to add something like `enum xrow_update_scalar_type type` to `struct xrow_update_string`, which is only used as member in `struct xrow_update_scalar`, which in turn already has `enum xrow_update_scalar_type type`. It seems to violate DRY principle, and I don't like it.
Perhaps we can create one scalar type `XUPDATE_TYPE_STRBIN` instead of `XUPDATE_TYPE_STR/XUPDATE_TYPE_BIN` pair, introduce new enum with `SUBTYPE_STR/SUBTYPE_BIN` values then rename `xrow_update_string` -> `xrow_update_strbin` and store the latter enum in it.

Reply:
I for the second variant too. May be we can start from just boolean flag instead of `XUPDATE_TYPE_STRBIN` or as you like.

Reply:
I've already done it in this way, please review.